packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -458,6 +458,9 @@ export class DefaultCrudRepository<
     data: DataObject<T>,
     options?: Options,
   ): Promise<void> {
+    if (id === undefined) {
```

Comment:
Can we use `id == null` to test both `null` and `undefined`?

Reply:
That is what I did initially, I was checking for both cases, but later when I tried calling updateById with null as id, it did not execute an updateAll, it simply returned EntityNotFoundError. Here is the output when updatById is called with a null id value;
`
{
  "error": {
    "statusCode": 404,
    "name": "Error",
    "message": "Entity not found: mymodel with id null",
    "code": "ENTITY_NOT_FOUND"
  }
}
`
Well you never know someone might have a nullable id field, with only one distinct null value in the database, and might want to update that one, so I changed my initial fix to check for only undefined values. Undefined is the real issue here, 
`(where as AnyObject)[idProp] = id;`
This causes an empty where statement when id is undefined, so all rows are updated as a result with the next updateAll statement;
`const result = await this.updateAll(data, where, options);`